pkg/client/imagestreamsecrets.go

```diff
@@ -0,0 +1,44 @@
+package client
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+// ImageStreamSecretsNamespacer has methods to work with ImageStreamSecret resources in a namespace
+type ImageStreamSecretsNamespacer interface {
+	ImageStreamSecrets(namespace string) ImageStreamSecretInterface
+}
+
+// ImageStreamSecretInterface exposes methods on ImageStreamSecret resources.
+type ImageStreamSecretInterface interface {
```

Comment:
It's a general secrets interface akin to Scale, but it's closer to image stream.  The primary reason is that the callers of ImageStreamSecrets are not coupled to ImageStream callers.  So keeping the interface separate simplifies what people pass in to specific callers.  I prefer the `*Namespacers` to  align with their callers, not with their logical grouping.